userspace/libsinsp/container.cpp

```diff
@@ -270,7 +270,10 @@ bool sinsp_container_manager::container_to_sinsp_event(const std::string& json,
 		// This can happen at startup when containers are
 		// being created as a part of the initial process
 		// scan.
-		scapevt->ts = sinsp_utils::get_current_time_ns();
+		// note: the following is only a convention and
+		// the timestamp will be setted when the event
+		// will be popped.
+		scapevt->ts = (uint64_t) - 1;
```

Comment:
So, when the event is stored to a scap file, and then the scap file gets replied, we assign to the event the current timestamp, ie: the time the scap file is being reproduced?

Reply:
This is invoked only in live mode when generating actual container events, for which it makes sense to just set the -1 value and let sinsp set the actual timestamp.

It is also invoked when dumping the containers in the initial state-of-the-machines scap block. In that case, we should get the current timestamp (or latest ts from previous events like before).

The two cases should be handled distinctly IMO.

This should easily be achievable by changing the container dump section like:

```cpp
if(container_to_sinsp_event(container_to_json(*it.second), &evt, it.second->get_tinfo(m_inspector)))
{
	evt.m_pevt->ts = (m_lastevent_ts == 0)
		? sinsp_utils::get_current_time_ns()
		: m_lastevent_ts; // to be added in the codebase
	dumper.dump(&evt);
}
```